cpp/src/arrow/array/validate.cc

```diff
@@ -637,6 +638,23 @@ struct ValidateArrayFullImpl {
 
 ARROW_EXPORT
 Status ValidateArrayFull(const ArrayData& data) {
+  if (data.null_count != -1) {
+    int64_t actual_null_count;
+    if (HasValidityBitmap(data.type->id()) && data.buffers[0]) {
```

Comment:
I notice ValidateArray doesn't check for `data.type == nullptr`, is that worth adding in this PR?

Reply:
Can do!